plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/ElkUtil.java

```diff
@@ -1021,6 +1019,42 @@ public static String debugFolderPath(final String subfolder) {
 
         return null;
     }
+    
+    /**
+     * Takes the given name and makes it safe to be used as a file or folder name. To do so, we replace all spaces by
+     * underscores and everything that is neither digit not standard character by hyphens.
+     * 
+     * @param name the name to convert to a proper path name.
+     * @return the proper path name.
+     */
+    public static String toSafePathName(final String name) {
```

Comment:
Performance. We'd need to compile regular expressions, call `replaceAll(...)` twice, etc. I do concede, however, that it would make the code much more readable, and since it's only used in debug mode, readability might in fact be more important than performance. I'm going to change it.